survey/src/main/java/com/quickbirdstudios/surveykit/backend/address/YandexAddressSuggestionProvider.kt

```diff
@@ -0,0 +1,85 @@
+package com.quickbirdstudios.surveykit.backend.address
+
+import com.quickbirdstudios.surveykit.AnswerFormat
+import java.io.FileNotFoundException
+import java.net.HttpURLConnection
+import java.net.URL
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+class YandexAddressSuggestionProvider(
+    private val apiKey: String,
+    private val resultsCount: Int = 5,
+    private val lang: String = "tr_TR",
+    onSuggestionListReady: (suggestions: List<AddressSuggestion>) -> Unit? = {}
+) : AddressSuggestionProvider(onSuggestionListReady) {
+
+    private fun getApiPath(apiKey: String, query: String, results: Int, lang: String) =
+        "https://geocode-maps.yandex.ru/1.x/?apikey=$apiKey" +
+            "&format=json&geocode=$query" +
+            "&results=$results" +
+            "&lang=$lang"
+
+    override fun input(query: String) {
+        val apiPath =
+            getApiPath(apiKey = apiKey, query = query, results = resultsCount, lang = lang)
+
+        GlobalScope.launch(Dispatchers.Main) { // launches coroutine in main thread
```

Comment:
It seems more an operation for the `Default` dispatcher. Then you can switch to `Main` when you update the callback